src/ripple/app/paths/cursor/DeliverNodeForward.cpp

```diff
@@ -35,7 +35,8 @@ TER PathCursor::deliverNodeForward (
     AccountID const& uInAccountID,    // --> Input owner's account.
     STAmount const& saInReq,        // --> Amount to deliver.
     STAmount& saInAct,              // <-- Amount delivered, this invocation.
-    STAmount& saInFees) const       // <-- Fees charged, this invocation.
+    STAmount& saInFees,             // <-- Fees charged, this invocation.
+    bool callerHasLiquidity) const
```

Comment:
`callerHasLiquidity` is signifies that the caller (always a recursive call when true) already has liquidity. It's purpose is to prevent `advanceNode` from changing qualities when we've already gotten some liquidity at a better quality.